ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/git/GitServiceClientImpl.java

```diff
@@ -445,69 +419,60 @@ public void push(DevMachine devMachine,
-        if (remoteName != null) {
-            remoteListRequest.setRemote(remoteName);
-        }
-        String url = appContext.getDevMachine().getWsAgentBaseUrl() + REMOTE_LIST + "?projectPath=" + project.getPath();
-        asyncRequestFactory.createPostRequest(url, remoteListRequest).loader(loader).send(callback);
+        String params = "?projectPath=" + project.getPath() + (remoteName != null ? "&remoteName=" + remoteName : "") +
+                        "&verbose=" + String.valueOf(verbose);
+        String url = appContext.getDevMachine().getWsAgentBaseUrl() + REMOTE + params;
+        asyncRequestFactory.createGetRequest(url).loader(loader).send(callback);
     }
 
-    /** {@inheritDoc} */
     @Override
     public Promise<List<Remote>> remoteList(DevMachine devMachine, ProjectConfig project, @Nullable String remoteName, boolean verbose) {
-        RemoteListRequest remoteListRequest = dtoFactory.createDto(RemoteListRequest.class).withVerbose(verbose);
-        if (remoteName != null) {
-            remoteListRequest.setRemote(remoteName);
-        }
-        String url = devMachine.getWsAgentBaseUrl() + REMOTE_LIST + "?projectPath=" + project.getPath();
-        return asyncRequestFactory.createPostRequest(url, remoteListRequest)
+        String params = "?projectPath=" + project.getPath() + (remoteName != null ? "&remoteName=" + remoteName : "") +
+                        "&verbose=" + String.valueOf(verbose);
+        String url = appContext.getDevMachine().getWsAgentBaseUrl() + REMOTE + params;
+        return asyncRequestFactory.createGetRequest(url)
                                   .loader(loader)
                                   .send(dtoUnmarshallerFactory.newListUnmarshaller(Remote.class));
     }
 
     @Override
-    public Promise<List<Remote>> remoteList(DevMachine devMachine, Path project, String remote, boolean verbose) {
-        RemoteListRequest remoteListRequest = dtoFactory.createDto(RemoteListRequest.class).withVerbose(verbose);
-        if (remote != null) {
-            remoteListRequest.setRemote(remote);
-        }
-        String url = devMachine.getWsAgentBaseUrl() + REMOTE_LIST + "?projectPath=" + project;
-        return asyncRequestFactory.createPostRequest(url, remoteListRequest)
+    public Promise<List<Remote>> remoteList(DevMachine devMachine, Path project, String remoteName, boolean verbose) {
+        String params = "?projectPath=" + project.toString() + (remoteName != null ? "&remoteName=" + remoteName : "") +
+                        "&verbose=" + String.valueOf(verbose);
+        String url = appContext.getDevMachine().getWsAgentBaseUrl() + REMOTE + params;
+        return asyncRequestFactory.createGetRequest(url)
                                   .loader(loader)
                                   .send(dtoUnmarshallerFactory.newListUnmarshaller(Remote.class));
     }
 
-    /** {@inheritDoc} */
     @Override
     public void branchList(DevMachine devMachine,
```

Comment:
Make sure that you add @Deprecated annotation to all methods for which you provide similar method with promise